app/controllers/api/base_controller.rb

```diff
@@ -0,0 +1,49 @@
+module Api
+  class BaseController < ApplicationController
+    protect_from_forgery with: :null_session
+
+    skip_before_action :set_current_team
+    skip_before_action :run_first_time_setup
+    skip_before_action :ensure_team_exists_for_subdomain!
+    skip_before_action :require_login
+    skip_before_action :user_belongs_to_team!
+
+    before_action :require_api_token
+    before_action :set_current_team
+
+    private
+
+    def render_errors(object)
+      errors = object.errors.messages.map do |field, error_messages|
+        {field: field, errors: error_messages}
+      end
+
+      render json: {message: "Validation Failed", errors: errors}, status: 422
+    end
+
+    def require_api_token
+      request_http_token_authentication unless current_token
+    end
+
+    def current_token
+      @current_token ||=
+        authenticate_with_http_token do |token, _options|
```

Comment:
Had no idea rails had [built in helpers for token authentications](http://api.rubyonrails.org/classes/ActionController/HttpAuthentication/Token.html)